ddsrecorder_participants/src/cpp/recorder/mcap/McapHandler.cpp

```diff
@@ -906,6 +1010,8 @@ void McapHandler::update_channels_nts_(
 
             assert(channel.first.m_topic_name == channel.second.topic);
             mcap::Channel new_channel(channel.second.topic, "cdr", new_schema_id, channel.second.metadata);
+            // Check if there is enough space available to write the channel
+            mcap_size_tracker_.check_and_increase_mcap_size(new_channel, mcap::internal::KeyValueMapSize(new_channel.metadata));
```

Comment:
I'd try to avoid creating and passing a metadata object, as at least now it should be constant.